### PR TITLE
owner(ticdc): integrate scheduler(v2) with the Owner (Part I)

### PR DESCRIPTION
**Pull request by liuzix**
**Notes for reviewers: This PR moved the original content of `cdc/owner/scheduler.go` to `cdc/owner/scheduler_v1.go`.**

### What problem does this PR solve? 
- We need to integration the new scheduler with the Owner smoothly.
- We need to prepare for a "rollback" plan in case we cannot release the new scheduler in time.

### What is changed and how it works?
- Made `scheduler` an interface, and made it compatible with the design of the new scheduler.
- Implemented `schedulerV1CompatWrapper`, which wraps the old scheduler and exposes the `scheduler` interface.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change

Side effects

 - Increased code complexity

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/owner/changefeed.go [hunk 1]
```diff
@@ -178,14 +180,20 @@ func (c *changefeed) tick(ctx cdcContext.Context, state *orchestrator.Changefeed
 		// So we return here.
 		return nil
 	}
-	shouldUpdateState, err := c.scheduler.Tick(c.state, c.schema.AllPhysicalTables(), captures)
+	newCheckpointTs, newResolvedTs, err := c.scheduler.Tick(c.state, c.schema.AllPhysicalTables(), captures)
 	if err != nil {
 		return errors.Trace(err)
 	}
-	if shouldUpdateState {
+	if newCheckpointTs != schedulerv2.CheckpointCannotProceed {
```
Comment (nongfushanquan) [inline cdc/owner/changefeed.go:187]: what does it mean

Comment (liuzix) [reply cdc/owner/changefeed.go:187]: I added some comments

**Review by asddongmen — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Pull request merged**